critiquebrainz/db/avg_rating.py

```diff
@@ -0,0 +1,118 @@
+import sqlalchemy
+from critiquebrainz import db
+from critiquebrainz.db import exceptions as db_exceptions
+
+
+def update(entity_id, entity_type):
+    """Updates the average rating of the entity
+
+    It does so by collecting all the review_ids for a given entity from review
+    and selecting respective revisions with latest timestamp
+    and calculating sum and count of ratings
+
+    Args:
+        entity_id (uuid): ID of the entity
+        entity_type (str): Type of the entity
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            WITH LatestRevisions AS (
+                SELECT review_id,
+                       MAX("timestamp") created_at
+                  FROM revision
+                 WHERE review_id in ( 
+                           SELECT id 
+                             FROM review
+                            WHERE entity_id = :entity_id
+                              AND entity_type = :entity_type)
+              GROUP BY review_id)
+            
+            SELECT SUM(rating),
+                   COUNT(rating)
+              FROM revision
+        INNER JOIN LatestRevisions
+                ON revision.review_id = LatestRevisions.review_id
+               AND revision.timestamp = LatestRevisions.created_at
+        """), {
```

Comment:
No. Because revisions with no rating will not make any difference to the sum and count.